File: aurafit-frontend/src/Models/ProductVariantImage.ts
import { ProductVariant } from "./ProductVariant";

export class ProductVariantImage {
    id: number;
    imageUrl: string;
    variant:  ProductVariant;

    constructor(id: number, imageUrl: string, variant:  ProductVariant) {
        this.id = id;
        this.imageUrl = imageUrl;
        this.variant =  variant;
    }
}

File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/Beans/ProductVariantImage.java
package app.aurafitbackend.Beans;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "product_variants_images")
@NoArgsConstructor
@AllArgsConstructor
@Data
public class ProductVariantImage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;
    private String imageUrl;

    @ManyToOne
    @JsonIgnore
    private ProductVariant variant;
}
